include/envoy/tracing/http_tracer.h

```diff
@@ -36,7 +36,16 @@ class Driver {
 public:
   virtual ~Driver() {}
 
-  virtual SpanPtr startSpan(const std::string& operation_name, SystemTime start_time) PURE;
+  /**
+   * Start driver specific span.
+   */
+  virtual SpanPtr startSpan(const std::string& parent_context, const std::string& operation_name,
+                            SystemTime start_time) PURE;
+
+  /**
+   * Inject span context into HTTP carrier.
+   */
+  virtual void inject(Span* active_span, Http::HeaderMap& headers) PURE;
```

Comment:
I don't understand why this needs to exist here. You only ever call inject() right after you call startSpan(), so just do it all inside the startSpan() flow of calls.